liquidations/angle/index.ts

```diff
@@ -0,0 +1,102 @@
+import axios from "axios";
+import { gql, request } from "graphql-request";
+import { ethers } from "ethers";
+import { providers } from "../utils/ethers";
+import BigNumber from "bignumber.js";
+import { getPagedGql } from "../utils/gql";
+
+const subgraphUrl = "https://api.thegraph.com/subgraphs/name/picodes/borrow";
+
+const vaultDataQuery = gql`
+query vaultDatas {
+    vaultDatas(where: {isActive: true}) {
+      id
+      collateralAmount
+      vaultManager {
+        agToken
+        collateral
+        collateralFactor
+      }
+      owner
+    }
+  }
+`;
+
+type VaultData = {
+    id: string;
+    collateralAmount: BigNumber;
+    vaultManager: {
+        agToken: string;
+        collateral: string;
+        collateralFactor: BigNumber;
+    }
+};
+
+
+const getVaultData = async () => {
+    const vaultData = (await getPagedGql(subgraphUrl, vaultDataQuery, "vaultDatas")) as VaultData[];
+    return vaultData;
+};
+
+const getPrices = async () => {
+  const url: string = 'https://api.coingecko.com/api/v3/simple/price?ids=ageur&vs_currencies=usd';
+  const price = (await axios.get(url)).data.ageur.usd;
+  return price
+} 
+
+// returns vault agEUR debt in $
+const getVaultDebt = async (id: string) => {
+  const vaultManager = id.split("_")[0]
+  const vaultId = id.split("_")[1]
+  const vaultManagerContract = new ethers.Contract(
+    vaultManager,
+    ["function getVaultDebt(uint256) view returns (uint256)"],
+    providers.ethereum
+  );
+  const vaultDebtRaw = (await vaultManagerContract.getVaultDebt(vaultId));
+  // convert vault debt to $
+  const vaultDebt = (await getPrices() * vaultDebtRaw); 
+  return vaultDebt.toString();
+}
+
+const EXPLORER_BASE_URL = "https://etherscan.io/address/"
+
+const positions = async () => {
+  const vaultData: any[] = await getVaultData();
+
+  const positions : {owner: string, liquidationPrice: number, collateral: string, collateralAmount: string, extra: any}[] = [];
+  for (const vault of vaultData) {
+    const owner = vault.owner;
+    var collateral = vault.vaultManager.collateral;
+    const collateralAmount = vault.collateralAmount;
+
+    // liquidation price computation
+    const vaultDebt = await getVaultDebt(vault.id);
+    const collateralFactor = parseFloat(vault.vaultManager.collateralFactor) / 10e8
+    let liquidationPrice: number
+    if (collateral == '0x2260fac5e5542a773aa44fbcfedf7c193bc2c599') {
+      liquidationPrice = parseFloat(vaultDebt) / (parseFloat(collateralAmount)*10e9*collateralFactor)
```

Comment:
Is WBTC the only token which needed converting decimals with `*10e9` and why `*10e9`? If there could be more assets which needs conversion, we prefer to read the decimals from on-chain:

```ts
const { output: decimals } = await sdk.api.erc20.decimals(token, chain);
```
... or use the results from llama token price API: https://github.com/DefiLlama/DefiLlama-Adapters/blob/de5be4926982d43e875d164e7f56676d0a644822/liquidations/utils/compound-helpers.ts#L41

Reply:
Or you can use this piece of code to get the price from DefiLlama Token Price API

```ts
type Price = {
  decimals: number;
  symbol: string;
  price: number;
  timestamp: number;
  confidence: number;
};

// this code will return a dictionary shaped as the Price type
const getAgeurPriceInUsd = async () => {
  const tokenId = "ethereum:0x1a7e4e63778b4f12a199c062f3efdd288afcbce8";
  const res = (
    await axios.get("https://coins.llama.fi/prices/current/" + tokenId)
  ).data.coins[tokenId] as Price;
  return res;
};

```